### PR TITLE
feat: payment provider and customer step in checkout

### LINKED ISSUE #56: [Feature]: develop/finish payment mock and order checkout functionality
### How the project can be improved?

For now we have almost full checkout process done. Last part is to develop payment provider mock and checkout process fulfillment.

### What are the acceptance criteria?

- [ ]

### Additional information

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #57: [Feature]: Add step in checkout about customer data
### How the project can be improved?

https://www.vendure.io/docs/graphql-api/shop/mutations/#setcustomerfororder

### What are the acceptance criteria?

- [ ]

### Additional information

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Baroshem**
## Description

Closes #56 
Closes #57 

I decided to release these two functionalities as one purely because payment wont work without customer step.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by michaelbromley — COMMENTED** [review]
The queries look good to me.

File: packages/api-client/src/api/setCustomerForOrder/index.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    customQuery, { setCustomerForOrder: { query: setCustomerForOrderMutation, variables: setCustomerForOrderVariables } }
+  );
+
+  const request = await context.client.mutate({
+    mutation: gql`${setCustomerForOrder.query}`,
+    variables: setCustomerForOrder.variables,
+    fetchPolicy: NO_CACHE_FETCH_POLICY
+  }) as SetCustomerForOrderResponse;
+
+  return request;
```
Comment (Mateuszpietrusinski) [inline packages/api-client/src/api/setCustomerForOrder/index.ts:22]: You can simplify that.
```suggestion
  return await context.client.mutate({
    mutation: gql`${setCustomerForOrder.query}`,
    variables: setCustomerForOrder.variables,
    fetchPolicy: NO_CACHE_FETCH_POLICY
  }) as SetCustomerForOrderResponse;
```

File: packages/api-client/src/api/getPaymentMethods/index.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  const { eligiblePaymentMethods } = context.extendQuery(
+    customQuery, { eligiblePaymentMethods: { query: eligiblePaymentMethodsQuery, variables: getPaymentMethodsVariables } }
+  );
+
+  const request = await context.client.query<RequestDataStructure<'eligiblePaymentMethods', PaymentMethodQuote[]>>({
+    query: gql`${eligiblePaymentMethods.query}`,
+    variables: eligiblePaymentMethods.variables,
+    fetchPolicy: NO_CACHE_FETCH_POLICY
+  });
+  return request;
```
Comment (Mateuszpietrusinski) [inline packages/api-client/src/api/getPaymentMethods/index.ts:19]:
```suggestion
  return await context.client.query<RequestDataStructure<'eligiblePaymentMethods', PaymentMethodQuote[]>>({
    query: gql`${eligiblePaymentMethods.query}`,
    variables: eligiblePaymentMethods.variables,
    fetchPolicy: NO_CACHE_FETCH_POLICY
  });
```

File: packages/api-client/src/api/setPaymentMethod/index.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    customQuery, { addPaymentToOrder: { query: addPaymentToOrderMutation, variables: setPaymentMethodVariables } }
+  );
+
+  const request = await context.client.mutate({
+    mutation: gql`${addPaymentToOrder.query}`,
+    variables: addPaymentToOrder.variables,
+    fetchPolicy: NO_CACHE_FETCH_POLICY
+  }) as SetShippingMethodResponse;
+
+  return request;
```
Comment (Mateuszpietrusinski) [inline packages/api-client/src/api/setPaymentMethod/index.ts:27]: Redundant variable
```suggestion
  return await context.client.mutate({
    mutation: gql`${addPaymentToOrder.query}`,
    variables: addPaymentToOrder.variables,
    fetchPolicy: NO_CACHE_FETCH_POLICY
  }) as SetShippingMethodResponse;
```

File: packages/api-client/src/api/transitionOrderToState/index.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    customQuery, { transitionOrderToState: { query: transitionOrderToStateMutation, variables: transitionOrderToStateVariables } }
+  );
+
+  const request = await context.client.mutate({
+    mutation: gql`${transitionOrderToState.query}`,
+    variables: transitionOrderToState.variables,
+    fetchPolicy: NO_CACHE_FETCH_POLICY
+  }) as TransitionOrderToState;
+
+  return request;
```
Comment (Mateuszpietrusinski) [inline packages/api-client/src/api/transitionOrderToState/index.ts:22]:
```suggestion
  return await context.client.mutate({
    mutation: gql`${transitionOrderToState.query}`,
    variables: transitionOrderToState.variables,
    fetchPolicy: NO_CACHE_FETCH_POLICY
  }) as TransitionOrderToState;
```

File: packages/theme/nuxt.config.js [hunk 5]
```diff
@@ -132,6 +134,12 @@ export default {
       })
     ]
   },
+  router: {
+    extendRoutes(routes) {
+      getRoutes(`${__dirname}/_theme`)
+        .forEach((route) => routes.unshift(route));
+    }
+  },
```
Comment (Mateuszpietrusinski) [inline packages/theme/nuxt.config.js:142]: Is that neccessary?

**Review by Mateuszpietrusinski — APPROVED** [review]
Good job 👏

File: packages/theme/nuxt.config.js (hunk 5 above)
Comment (Baroshem) [reply packages/theme/nuxt.config.js:142]: Yes, I need to have a checkout flow start from customer not shipping and in order to do that I have to extend default routes.

**Pull request merged**